core/src/main/scala/io/projectglow/transformers/normalizevariants/VariantNormalizer.scala

```diff
@@ -76,92 +76,106 @@ private[projectglow] object VariantNormalizer extends GlowLogging {
-              }
-
-            val isFromSplit = vcStreamAfterMaybeSplit.length > 1
-
-            val vcStreamAfterMaybeNormalize = if (doNormalize) {
-              vcStreamAfterMaybeSplit.map(vc =>
-                VariantNormalizer.normalizeVC(vc, refGenomeDataSource.get))
-            } else {
-              vcStreamAfterMaybeSplit
-            }
-
-            vcStreamAfterMaybeNormalize.map(
-              variantContextToInternalRowConverter.convertRow(_, isFromSplit)
-            )
-
-          case None => Stream(row)
+        it.map { row =>
+          val isFromSplit = false
+          internalRowToVariantContextConverter.convert(row) match {
 
+            case Some(vc) =>
+              variantContextToInternalRowConverter
+                .convertRow(VariantNormalizer.normalizeVC(vc, refGenomeDataSource.get), isFromSplit)
+            case None => row
+          }
         }
       }
+    } else {
+      dfAfterMaybeSplit.queryExecution.toRdd
     }
 
-    SQLUtils.internalCreateDataFrame(df.sparkSession, mapped, schema, false)
-
+    SQLUtils.internalCreateDataFrame(df.sparkSession, rddAfterMaybeNormalize, schema, false)
   }
 
   /**
-   * Splits a vc to bi-alleleic using gatk splitting function
+   * Generates a new DataFrame by splitting the variants in the input DataFrame
    *
-   * @param vc
+   * @param inputDf
    * @return a Stream of split vc's
    */
-  private def splitVC(vc: VariantContext): Stream[VariantContext] = {
-    GATKVariantContextUtils
-      .splitVariantContextToBiallelics(
-        vc,
-        false,
-        GenotypeAssignmentMethod.BEST_MATCH_TO_ORIGINAL,
-        false
-      )
-      .asScala
-      .toStream
+  def splitVariants(inputDf: DataFrame): DataFrame = {
+
+    val variantDf = if (inputDf.schema.fieldNames.contains("attributes")) {
+      // TODO: What to do when INFO fields are not flattened?
+      inputDf // WIP
+    } else {
+      inputDf
+    }
+
+    // Add/replace splitFromMultiAllelic column and explode alternateAlleles column
+    val dfAfterAltAlleleSplit = variantDf
+      .withColumn(
+        splitFromMultiAllelicField.name,
+        when(size(col(alternateAllelesField.name)) > 1, true).otherwise(false))
+      .select(
+        col("*"),
+        posexplode(col(alternateAllelesField.name)).as(Array("altAllelePos", "splitAlleles")))
+
+    // Seq of functions to be applied on ArrayTyped INFO fields to split them if necessary
+    val infoFieldsSplittingFunctions: Seq[DataFrame => DataFrame] = variantDf
+      .schema
+      .filter(
+        field =>
+          field.name.startsWith(infoFieldPrefix) &&
+          (field.dataType match {
+            case ArrayType(_, _) => true
+            case _ => false
+          }))
+      .map(field =>
+        (f: DataFrame) => {
+          f.withColumn(
+            field.name,
+            when(
+              size(col(field.name)) === size(col(alternateAllelesField.name)),
+              expr(s"${field.name}[altAllelePos]")))
```

Comment:
I don't think this is necessarily correct? If the number is `A` then it makes sense, but if the number is `R` then there's an entry for the reference allele as well.